src/modules/editor/sagas.ts

```diff
@@ -205,3 +213,20 @@ function* handleDropItem(action: DropItemAction) {
     yield put(addItem(asset, position))
   }
 }
+
+function* handleScreenshot(_: ScreenshotAction) {
+  try {
+    const screenshot = yield call(() => editorWindow.editor.takeScreenshot())
+    if (screenshot) {
+      const thumbnail = yield call(() => resizeScreenshot(screenshot, 246, 182))
+      if (thumbnail) {
```

Comment:
If this is false, shouldn't we retry?

Reply:
These `if`s shouldn't be false unless the canvas has been detached/unmounted, in which case we can't take a screenshot, so i think it's not worth it implementing a retry logic. For most cases this should work fine.

Reply:
Perfect